templates/cluster-template-topology.yaml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: Cluster
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: '${CLUSTER_NAME}'
+  name: '${CLUSTER_NAME}'
+  namespace: '${NAMESPACE}'
+spec:
+  topology:
+    class: '${CLUSTER_CLASS_NAME}'
+    controlPlane:
+      replicas: ${CONTROL_PLANE_MACHINE_COUNT}
+    variables:
+    - name: sshKey
+      value: '${IONOSCLOUD_MACHINE_SSH_KEY}'
+    - name: controlPlaneIpAddr
+      value: ${CONTROL_PLANE_ENDPOINT_IP}
+    - name: controlPlanePort
+      value: ${CONTROL_PLANE_ENDPOINT_PORT:-6443}
+    - name: credsSecretName
+      value: '${CLUSTER_NAME}-credentials'
+    - name: datacenterID
+      value: '${IONOSCLOUD_DATACENTER_ID}'
+    - name: location
+      value: '${CONTROL_PLANE_ENDPOINT_LOCATION}'
+    - name: controlPlaneNumCores
+      value: '${IONOSCLOUD_MACHINE_NUM_CORES}'
+    - name: controlPlaneMemoryMB
+      value: '${IONOSCLOUD_MACHINE_MEMORY_MB}'
+    - name: controlPlaneImageID
+      value: '${IONOSCLOUD_MACHINE_IMAGE_ID}'
+    - name: workerNumCores
+      value: '${IONOSCLOUD_MACHINE_NUM_CORES}'
+    - name: workerMemoryMB
+      value: '${IONOSCLOUD_MACHINE_MEMORY_MB}'
+    - name: workerImageID
+      value: '${IONOSCLOUD_MACHINE_IMAGE_ID}'
+    version: '${KUBERNETES_VERSION}'
+    workers:
+      machineDeployments:
+      - class: ${CLUSTER_CLASS_NAME}-worker
```

Comment:
Your naming for the MachineDeployement `worker` is different in the clusterclass-template. If I try to run the commands it will let me know that a MachineDeployment with the name `xxx-worker` was not found in ClusterClass XY